tests: Conditionally compile MKFS Fat FS test

decsny opened the pull request:
This is a hotfix to work around build issues due to this FS test using the flash API innappropriately.

de-nordic reviewed (changes requested):
This comments ifdefs entire unit, just use CMakeLists if to do this.

carlescufi commented:
> This comments ifdefs entire unit, just use CMakeLists if to do this.



> This comments ifdefs entire unit, just use CMakeLists if to do this.

`*_sources_ifdef(CONFIG_X	x.c)`

Commit:
tests: fat_fs_api: Conditionally compile MKFS test

MKFS testing inappropriately uses flash API in the FAT FS API test,
causing build errors for the test when testing file systems on storage
devices other than flash. Hotfix by making this part of the test
conditional on CONFIG_FLASH.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

decsny commented:
> @decsny Can you explain the problem a little bit? How to reproduce the problem?

If you are using the fat fs on a non flash device there will be a build error. The test is for fat fs not for flash, so there should not be flash functions in the test without some compilation guard.

de-nordic commented:
> > @decsny Can you explain the problem a little bit? How to reproduce the problem?
> 
> If you are using the fat fs on a non flash device there will be a build error. The test is for fat fs not for flash, so there should not be flash functions in the test without some compilation guard.

You mean that when you build for FAT FS via disk access on non-flash device?

decsny commented:
> > > @decsny Can you explain the problem a little bit? How to reproduce the problem?
> > 
> > 
> > If you are using the fat fs on a non flash device there will be a build error. The test is for fat fs not for flash, so there should not be flash functions in the test without some compilation guard.
> 
> You mean that when you build for FAT FS via disk access on non-flash device?

when you build this fat_fs_api test for a non flash device yes

de-nordic reviewed (approved):
OK for hotfix. 
Please log issue in Zephyr for the test to be fixed for non-CONFIG_FLASH devices.

stephanosio commented:
Related issue: https://github.com/zephyrproject-rtos/zephyr/pull/53021
